modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/endpoint/AbstractAssetManagerRestEndpoint.java

```diff
@@ -310,11 +311,12 @@ public Response getAsset(@PathParam("mediaPackageID") final String mediaPackageI
             }
           }
 
-          final String fileName = mediaPackageElementID
-                  .concat(".")
-                  .concat(asset.getMimeType().bind(suffix).getOr("unknown"));
+          if (StringUtils.isEmpty(fileName)) {
```

Comment:
Haven't actually run the code, but looks good.
Minor suggestion, also prevent blank file names like "`  `".

```suggestion
          if (StringUtils.isBlank(fileName)) {
```